lib/animina/checks/read_profile_check.ex

```diff
@@ -37,19 +38,47 @@ defmodule Animina.Checks.ReadProfileCheck do
     end
   end
 
-  defp user_can_view_profile(true, true, _) do
+  defp check_if_user_can_view_profile(_actor, _params, _) do
+    true
+  end
+
+  defp user_can_view_profile(true, :normal) do
+    true
+  end
+
+  defp user_can_view_profile(true, :under_investigation) do
     true
   end
 
-  defp user_can_view_profile(false, true, points) when points >= 10 do
+  defp user_can_view_profile(false, :under_investigation) do
+    false
+  end
+
+  defp user_can_view_profile(_, _) do
     true
   end
 
-  defp user_can_view_profile(false, false, points) when points >= 20 do
+  defp user_can_view_profile(true, :normal_, _, _) do
     true
   end
 
-  defp user_can_view_profile(_, _, _) do
+  defp user_can_view_profile(true, :under_investigation_, _, _) do
+    true
+  end
+
+  defp user_can_view_profile(false, :under_investigation, _, _, _) do
+    false
+  end
+
+  defp user_can_view_profile(false, _, false, true, points) when points >= 10 do
```

Comment:
The needed points shouldn't be hard coded here. We have this point priceless module which should be used to define how many points are needed for what to do. That is our central point of defining the needed points for anything. I want that in one place. Otherwise we have it scattered around all over the place. Hard to manage.

Reply:
This has been changed, I now do it from the Actions Price List Module we had prior